s3-scan-tar/scanner.py

```diff
@@ -36,28 +38,28 @@ def __init__(self, filehandle, maxsize=None):
     def write(self, s) -> int:
         raise NotImplementedError("write() not supported")
 
-    def read(self, read_length) -> bytes:
+    def read(self, read_length: Any) -> bytes:
         if self.maxsize:
             if not read_length:
                 # no lenght given, not sure we if can read this without going over the limit
                 raise NotImplementedError("Unlimited read requested. Not supported")
-            if self.filehandle.tell() + read_length >= self.maxsize:
+            if self.filehandle.tell() + read_length > self.maxsize:
                 logging.warning(
                     "Restricting file object and discarding the rest of the contents")
                 self.restricted = True
                 self.read_flush()
-                return None
+                return b''
         return self.filehandle.read(read_length)
 
     def read_flush(self):
         """ Read and discard the rest of the file. """
-        while len(self.filehandle.read(1024**2)) > 0:  # read big chunks...
+        while len(self.filehandle.read(1024 ** 2)) > 0:  # read big chunks...
             pass
 
     def tell(self):
         return self.filehandle.tell()
 
-    def seek(self, pos):
+    def seek(self, pos: Any, **kwargs) -> int:
```

Comment:
```suggestion
    def seek(self, pos: int, **kwargs) -> int:
```
Eller misforstår jeg noe?